WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java

```diff
@@ -913,25 +930,17 @@ public void onResponse(JSONObject response) {
                     if (errorReason != null && errorReason.equals(REASON_ERROR_TAKEN) && mListener != null) {
                         mListener.onMagicLinkRequestSuccess(mUsername);
                     } else {
-                        if (isUsernameEmail()) {
-                            showSelfHostedSignInForm();
-                        } else {
-                            showPasswordFieldAndFocus();
-                        }
+                        onWPComEmailCheckError(false);
                     }
                 } catch (JSONException error) {
                     AppLog.e(AppLog.T.MAIN, error);
-                    if (isUsernameEmail()) {
-                        showSelfHostedSignInForm();
-                    } else {
-                        showPasswordFieldAndFocus();
-                    }
+                    onWPComEmailCheckError(false);
                 }
             }
         }, new RestRequest.ErrorListener() {
             @Override
             public void onErrorResponse(VolleyError error) {
-                showPasswordFieldAndFocus();
+                onWPComEmailCheckError(true);
```

Comment:
In case of errors, even a classic network error, we're calling `onWPComEmailCheckError(true);` that shows the password field and tries to give the focus to it. The username field is left unchanged. We probably need to check the kind of error here, and change the response according to it.